packages/modules/itwin-viewer-react/src/components/iModel/IModelLoader.tsx

```diff
@@ -226,7 +226,31 @@ const Loader: React.FC<ModelLoaderProps> = React.memo(
       getModelConnection().catch((error) => {
         errorManager.throwFatalError(error);
       });
-    }, [contextId, iModelId, snapshotPath, frontstages, backstageItems]);
+    }, [
+      contextId,
+      iModelId,
+      changeSetId,
+      snapshotPath,
+      frontstages,
+      backstageItems,
+      blankConnection,
+      blankConnectionViewState,
+    ]);
+
+    useEffect(() => {
+      const closeIModelConnection = async () => {
+        const iModelConn = UiFramework.getIModelConnection();
+        if (iModelConn) {
+          await iModelConn.close();
+        }
+      };
+
+      return () => {
```

Comment:
did this not work as cleanup in the current useEffect that opens the connection?

Reply:
Did it in a separate one in order to avoid closing connections when the frontstages, backstageItems, and blankConnectionViewState changed

Reply:
isn't that desired though? All of those items (except backstageItems, that could probably be removed from the deps list for that effect) could result in a closed connection....for example, one of the frontstages requires a connection, so one is created....frontstages changes and now none of them require connection....this would leave an orphaned connection no? Similar case with the blankConnectionViewState...that is only used when establishing a connection, so if it changes wouldn't we want to close and re-open a new one?

Reply:
also, in all these cases a new connection is being opened whether we close the existing one or not.....we should probably add a few more checks there, but theoretically all of those deps could result in a new and/or closed connection

Reply:
Good catch, updated as we talked about.